engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java

```diff
@@ -602,6 +603,49 @@ public void suspendedArgumentsUnappliedFunction() throws Exception {
     }
   }
 
+  @Test
+  public void andConversions() throws Exception {
+    final URI uri = new URI("memory://and_conv.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import all
+
+    type Plus
+        Impl value dict
+
+        + self (that:Plus) = if self.dict != that.dict then Panic.throw "panic!" else
+          self.dict.plus self.value that.value
+    type Mul
+        Impl value dict
+
+        * self (that:Mul) = if self.dict != that.dict then Panic.throw "panic!" else
+          self.dict.mul self.value that.value
+
+    compute (a : Plus & Mul) (b : Plus & Mul) =
```

Comment:
A function that consumes two values that implement `Plus` and `Mul` _type classes_.

Reply:
More specifically, two values for which both conversion methods to `Plus` and `Mul` exist.

Reply:
And then it also does a runtime check to verify that both values have the same dictionary: `self.dict` `that.dict` - e.g. implementation.